velox/functions/sparksql/String.h

```diff
@@ -902,4 +902,87 @@ struct TranslateFunction {
   }
 };
 
+template <typename T>
+struct ConvFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  static constexpr bool is_default_ascii_behavior = true;
+
+  static const uint64_t kMaxUnsignedInt64_ = 0xFFFFFFFFFFFFFFFF;
+  static const int kMinBase = 2;
+  static const int kMaxBase = 36;
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& input,
+      int32_t fromBase,
+      int32_t toBase) {
+    if (input.empty()) {
+      return false;
+    }
+
+    // Consistent with spark, only supports fromBase belonging to [2, 36]
+    // and toBase belonging to [2, 36] or [-36, -2].
+    if (fromBase < kMinBase || fromBase > kMaxBase ||
+        std::abs(toBase) < kMinBase || std::abs(toBase) > kMaxBase) {
+      return false;
+    }
+
+    // Ignore space.
+    int i = 0;
+    for (; i < input.size(); i++) {
+      if (input.data()[i] != ' ') {
+        break;
+      }
+    }
+    const bool isInputNegative = (input.data()[i] == '-');
+    // Skip negative symbol.
+    i = isInputNegative ? i + 1 : i;
+    uint64_t unsignedValue;
+    auto fromStatus = std::from_chars(
+        input.data() + i, input.data() + input.size(), unsignedValue, fromBase);
+    if (fromStatus.ec == std::errc::result_out_of_range) {
+      unsignedValue = kMaxUnsignedInt64_;
+    }
+
+    if (unsignedValue == 0) {
+      result.append("0");
+      return true;
+    }
+
+    int64_t signedValue;
+    if (isInputNegative) {
+      if (toBase < 0) {
+        signedValue = -unsignedValue;
```

Comment:
Don't we need to handle overflow?

Should we pass signedValue to std::from_chars?

Reply:
Overflow is allowed, no need to specially handle, like spark.

It's possible that unsigned number is a use case. So we skipped negative symbol and use unsigned variable to let std::from_chars handle.